app/actions/remote/thread.ts

```diff
@@ -194,3 +199,152 @@ export const updateThreadFollowing = async (serverUrl: string, teamId: string, t
         return {error};
     }
 };
+
+enum Direction {
+    Up,
+    Down,
+}
+
+async function fetchBatchThreads(
+    serverUrl: string,
+    teamId: string,
+    options: FetchThreadsOptions,
+    pages?: number,
+): Promise<{error: unknown; data?: Thread[]}> {
+    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+
+    if (!operator) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    // if we start from the begging of time (since = 0) we need to fetch threads from newest to oldest (Direction.Down)
+    // if there is another point in time, we need to fetch threads from oldest to newest (Direction.Up)
+    let direction = Direction.Up;
+    if (options.since === 0) {
+        direction = Direction.Down;
+    }
+
+    const currentUser = await getCurrentUser(operator.database);
+    if (!currentUser) {
+        return {error: 'currentUser not found'};
+    }
+
+    const {config} = await getCommonSystemValues(operator.database);
+    const data: Thread[] = [];
+
+    const fetchThreadsFunc = async (opts: FetchThreadsOptions) => {
+        let page = 0;
+        const {before, after, perPage = General.CRT_CHUNK_SIZE, deleted, unread, since} = opts;
+
+        page += 1;
+        const {threads} = await client.getThreads(currentUser.id, teamId, before, after, perPage, deleted, unread, since, false, config.Version);
+        if (threads.length) {
+            // Mark all fetched threads as following
+            for (const thread of threads) {
+                thread.is_following = true;
+            }
+
+            data.push(...threads);
```

Comment:
Should we consider appending or pre-pending depending on whether we are going up or down? Or them being ordered won't provide us any improvement?

Reply:
When we read threads we're sorting by lastReplyAt. Not sure if having an already sorted set has any benefit on the performance of `sortBy`. Then again we'll be adding threads outside of that order anyways when fetching single threads that is, or by fetching another team's threads.